sources/core/Stride.Core.CompilerServices/Analyzers/STRDIAG005ReadonlyMemberIsReferenceType.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Runtime.Serialization;
+using Stride.Core.CompilerServices.Common;
+
+namespace Stride.Core.CompilerServices.Analyzers;
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class STRDIAG005ReadonlyMemberIsReferenceType : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "STRDIAG005";
+    private const string Title = "ReadOnlyMemberIsReferenceType";
+    private const string MessageFormat = "The [DataMember] Attribute is applied to a read-only member '{0}' with a non allowed Type. Use a reference type instead, string isn't allowed either.";
```

Comment:
nit: `allowed` -> `supported`, `Type` -> `type`.
Second sentence could be:
`Only mutable reference types are supported for read-only members.`

Reply:
[45db198](https://github.com/stride3d/stride/pull/1864/commits/45db19884ffe4700690bd89fb596911a0df68fcc)